android/libraries/rib-router-navigator/src/main/kotlin/com/uber/rib/core/StackRouterNavigator.kt

```diff
@@ -16,24 +16,33 @@
 package com.uber.rib.core
 
 import androidx.annotation.IntRange
+import androidx.annotation.VisibleForTesting
 import com.uber.rib.core.RouterNavigatorEvents.Companion.instance
 import java.util.ArrayDeque
 import java.util.Locale
 
 /**
  * Simple utility for switching a child router based on a state.
  *
- * @param <StateT> type of state to switch on.
+ * @param hostRouter Host router
+ * @param forceRouterCaching Override [RouterNavigatorState.isCacheable] behavior
  */
-open class StackRouterNavigator<StateT : RouterNavigatorState>(private val hostRouter: Router<*>) : RouterNavigator<StateT> {
+open class StackRouterNavigator<StateT : RouterNavigatorState>
+@JvmOverloads
+constructor(
+  private val hostRouter: Router<*>,
+  // Enabled by default for safe migration to newer lib version
+  private val forceRouterCaching: Boolean = true
```

Comment:
I believe adding a default parameter in constructor makes it a binary-incompatible change. We should use two secondary constructors instead.

Reply:
This constructor has `@JvmOverloads` annotation which will generate single-parameter constructor for usages in java code. If I understand you comment correctly.